app/models/secret_storage.rb

```diff
@@ -50,6 +50,47 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    VAULT_SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(vault_path(key))
+      return false if result.nil?
+      result.data[:vault]
+    end
+
+    def self.write(key, data)
+      Vault.logical.write(vault_path(key), vault: data)
```

Comment:
I don't think we should write the complete hash into vault, that will make reading the value via consul / cli complicated